### DISCUSSION TITLE
Changing of POST_UP / POST_DOWN not possible

### DISCUSSION THREAD

**Discussion by Natriumpikant** [wg-easy/wg-easy#263] [Q&A] [answered]
Hello!

I am running wg-easy on an azure-vm and it mostly runs absolute smooth.

I have just one issue - when i try to ping the different clients within the vpn from the host machine i get an error because the host machin e does not route the traffic through the docker network adapter. But i need the host machine also this vpn network.

Following i tried to change the POST_UP parameters like that:
_docker run -d \
  --name=wg-easy \
  -e WG_HOST=XXXXX  \
  -e PASSWORD=XXXXX  \
  -e WG_DEFAULT_ADDRESS=XXXXX \
  -e WG_DEFAULT_DNS=XXXXX \
  -e WG_ALLOWED_IPS=0.0.0.0/0 \
  -e WG_POST_UP=iptables -A FORWARD -i %i -j ACCEPT; iptables -A FORWARD -o %i -j ACCEPT; iptables -t nat -A POSTROUTING -o eth0 -j MASQUERADE
  -e WG_POST_DOWN=iptables -D FORWARD -i %i -j ACCEPT; iptables -D FORWARD -o %i -j ACCEPT; iptables -t nat -D POSTROUTING -o eth0 -j MASQUERADE
  -v ~/.wg-easy:/etc/wireguard \
  -p 51820:51820/udp \
  -p 51821:51821/tcp \
  --cap-add=NET_ADMIN \
  --cap-add=SYS_MODULE \
  --sysctl="net.ipv4.conf.all.src_valid_mark=1" \
  --sysctl="net.ipv4.ip_forward=1" \
  --restart unless-stopped \
  weejewel/wg-easy_

Executing this code  always shows following error:
**unknown shorthand flag: 'A' in -A
See 'docker run --help'.
-bash: -e: command not found
-bash: -v: command not found**

Is there another way to change the POSTUP/DOWN parameters? Or another way to route the traffic through wg on the host machine?

Also tried that:
` ip route add 172.18.0.0/24 via 172.17.0.1`
But also get an error:
**PING 172.18.0.3 (172.18.0.3) 56(84) bytes of data.
From 172.17.0.1 icmp_seq=1 Destination Host Unreachable**

appreciate any help!
thanks in advance

**Comment by eandersons** [comment, marked as answer]
You should try to wrap the value for `WG_POST_UP` and `WG_POST_UP` in quotes (single or double), otherwise `docker run` understands `-e WG_POST_UP=iptables -A FORWARD ...` as an instruction to set environment variable `WG_POST_UP` with the value `iptables`, then process the shorthand flag `-A` with  the value `FORWARD` and so on...
`docker run` does not have a shorthand flag `-A`, hence the error message:

> unknown shorthand flag: 'A' in -A
> See 'docker run --help'.

**Reply by Natriumpikant** [reply]
Thanks for your answer :)

I already tried that, if i put quotes around it like following:
_-e WG_POST_DOWN="iptables -D FORWARD -i %i -j ACCEPT; iptables -D FORWARD -o %i -j ACCEPT; iptables -t nat -D POSTROUTING -o eth0 -j MASQUERADE"_

or 

_-e WG_POST_DOWN='iptables -D FORWARD -i %i -j ACCEPT; iptables -D FORWARD -o %i -j ACCEPT; iptables -t nat -D POSTROUTING -o eth0 -j MASQUERADE'_

i get the following error:

**"docker run" requires at least 1 argument.
See 'docker run --help'.**

**Usage:  docker run [OPTIONS] IMAGE [COMMAND] [ARG...]**

**Run a command in a new container
-bash: -e: command not found
-bash: -v: command not found**

**Reply by eandersons** [reply]
If you are trying to run a multilline command, then at the end of all lines (except the last one) ` \` must be added:

```shell
docker run -d \
  --name=wg-easy \
  -e WG_HOST=XXXXX \
  -e PASSWORD=XXXXX \
  -e WG_DEFAULT_ADDRESS=XXXXX \
  -e WG_DEFAULT_DNS=XXXXX \
  -e WG_ALLOWED_IPS=0.0.0.0/0 \
  -e WG_POST_UP='iptables -A FORWARD -i %i -j ACCEPT; iptables -A FORWARD -o %i -j ACCEPT; iptables -t nat -A POSTROUTING -o eth0 -j MASQUERADE' \
  -e WG_POST_DOWN='iptables -D FORWARD -i %i -j ACCEPT; iptables -D FORWARD -o %i -j ACCEPT; iptables -t nat -D POSTROUTING -o eth0 -j MASQUERADE' \
  -v ~/.wg-easy:/etc/wireguard \
  -p 51820:51820/udp \
  -p 51821:51821/tcp \
  --cap-add=NET_ADMIN \
  --cap-add=SYS_MODULE \
  --sysctl="net.ipv4.conf.all.src_valid_mark=1" \
  --sysctl="net.ipv4.ip_forward=1" \
  --restart unless-stopped \
  weejewel/wg-easy
```

Otherwise shell will try to interpret each line as a separate command.

**Reply by Natriumpikant** [reply]
@eandersons 
thanks for your help :)

the issue was the space between the second and third **;** and **iptables**.

this works fine:
`-e WG_POST_UP="iptables -A FORWARD -i %i -j ACCEPT;iptables -A FORWARD -o %i -j ACCEPT;iptables -t nat -A POSTROUTING -o eth0 -j MASQUERADE" \
`
` -e WG_POST_DOWN="iptables -D FORWARD -i %i -j ACCEPT;iptables -D FORWARD -o %i -j ACCEPT;iptables -t nat -D POSTROUTING -o eth0" \
`




Nontheless this doesnt fix my main issue - still cant access my wg-clients through the host, where wg is hosted in a docker container.
Still get the error:
**PING 172.18.0.2 (172.18.0.2) 56(84) bytes of data.**
**From 172.17.0.1 icmp_seq=1 Destination Host Unreachable**
Edit - fixed that by adding an separate installation of wireguard outside the dockerengine. On the host vm now runs wg as client and inside the dockerengine the wg-server for multiple clients.
